src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -294,54 +292,31 @@ private void executeWorker(Webhook webhook, Card card) {
     }
 
     private void addScmDetails() {
+        Set<User> users;
+        List<ChangeLogSet<ChangeLogSet.Entry>> sets;
+
         try {
-            if (run instanceof AbstractBuild) {
-                AbstractBuild build = (AbstractBuild) run;
-                factsBuilder.addCulprits(run.getResult(), build.getCulprits());
-
-                ChangeLogSet changeSet = build.getChangeSet();
-                Set<ChangeLogSet.AffectedFile> files = new HashSet<>();
-                Set<User> authors = new HashSet<>();
-
-                for (Object o : changeSet.getItems()) {
-                    ChangeLogSet.Entry entry = (ChangeLogSet.Entry) o;
-                    authors.add(entry.getAuthor());
-                    files.addAll(getAffectedFiles(entry));
-                }
+            users = (Set<User>) run.getClass().getMethod("getCulprits").invoke(run);
+            sets = (List<ChangeLogSet<ChangeLogSet.Entry>>) run.getClass().getMethod("getChangeSets").invoke(run);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            users = Collections.emptySet();
+            sets = Collections.emptyList();
+        }
 
-                factsBuilder.addDevelopers(authors);
-                factsBuilder.addNumberOfFilesChanged(files.size());
-            } else {
-                try {
-                    // newer Jenkins uses jenkins.scm.RunWithSCM interface so such casting is not needed
-                    Method getCulprits = run.getClass().getMethod("getCulprits");
-                    @SuppressWarnings("unchecked")
-                    Set<User> users = (Set<User>) getCulprits.invoke(run);
-                    factsBuilder.addCulprits(run.getResult(), users);
-
-                    Method getChangeSets = run.getClass().getMethod("getChangeSets");
-                    @SuppressWarnings("unchecked")
-                    List<ChangeLogSet<ChangeLogSet.Entry>> sets = (List<ChangeLogSet<ChangeLogSet.Entry>>) getChangeSets.invoke(run);
-                    Set<User> authors = new HashSet<>();
-                    Set<ChangeLogSet.AffectedFile> files = new HashSet<>();
-                    if (Iterables.all(sets, Predicates.instanceOf(ChangeLogSet.class))) {
-                        for (ChangeLogSet<ChangeLogSet.Entry> set : sets) {
-                            for (ChangeLogSet.Entry entry : set) {
-                                authors.add(entry.getAuthor());
-                                files.addAll(getAffectedFiles(entry));
-                            }
-                        }
+        factsBuilder.addCulprits(users);
+        if (sets.size() > 0) {
+            Set<User> authors = new HashSet<>();
```

Comment:
sometimes direct implementation sometimes empty collection - what about https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#emptySet() ?

Reply:
Not sure what you mean? We are filling this outself and AddDevelopers is already checking

Reply:
The way how you create new collections, sometimes with `new` operator, sometimes with `Collections` class

Reply:
There's very good reason for that.

Reply:
Also stating the obvious here: